cli/cmd/proxy/envoy/http.go

```diff
@@ -69,6 +73,72 @@ type Secret struct {
 	LastUpdated string
 }
 
+// CallLoggingEndpoint requests the logging endpoint from Envoy Admin Interface for a given port
+// more can be read about that endpoint https://www.envoyproxy.io/docs/envoy/latest/operations/admin#post--logging
+func CallLoggingEndpoint(ctx context.Context, portForward common.PortForwarder, params string) (map[string]string, error) {
+	params = parseParams(params)
```

Comment:
For separation of concerns, I think it would make sense to have a struct passed in for params instead of a string. I'm imagining something like this:

```go
type LoggingConfig struct {
    ...
}

func ParseLoggingParams(params string) LoggingConfig {
    ...
    return LoggingConfig {
        ...
    }
}

func UpdateLogLevel(ctx context.Context, portForward common.PortForwarder, cfg LoggingConfig) (map[string]string, error) {
    ...
}
````

Reply:
we could do that, but we'd have to have the `log` package call this function from the `envoy` package to parse the params into the struct, and then pass that struct into the CallLoggingEndpoint (I'm not 100% on this name tbh) function to transform it back into a string for the http call

I was kinda on the fence about where to put this parsing command, on the one hand it's specific to envoy, but thinking more about it having the parse know about what format the command takes seems like it knows a bit too much about it's consumers

Reply:
so for this I left it as a string and moved parsing up to the `loglevel` package, I'm gonna noodle a bit on making it a struct, the params for this call are a bit wonky so I wanna think of something that makes sense, for reference it's:
`?level={level}` to set the level on all loggers
`?{someLogger}:{level}` to set the level on one logger
`?paths={someLogger}:{level},{anotherLogger}:{maybeDifferentLevel}` to set the level on multiple loggers

I feel like _maybe_ the following could work:

```
type LoggerConfig struct {
      GlobalLevel string
      IndividualLevels map[string]string
}
```
and have the param parsing in the envoy package check if `GlobalLevel` is set and if so then just use that and ignore the `IndividualLevels` map, what do you think?

Reply:
ended up implementing this in `common/envoy/logger_params.go`